lib/iris/cube.py

```diff
@@ -1632,7 +1637,38 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
-        self._dtype = dtype
+        if dtype != self.dtype:
+            if dtype is not None:
+                if not self.has_lazy_data():
+                    emsg = 'Cube does not have lazy data, cannot set dtype.'
+                    raise ValueError(emsg)
+                dtype = np.dtype(dtype)
+                if dtype.kind != 'i':
+                    emsg = ('Can only cast lazy data to integral dtype, '
+                            'got {!r}.')
+                    raise ValueError(emsg.format(dtype))
+                self._fill_value = None
+            self._dtype = dtype
+
+    @property
+    def fill_value(self):
+        return self._fill_value
+
+    @fill_value.setter
+    def fill_value(self, fill_value):
+        if fill_value is not None:
+            # Convert the given value to the dtype of the cube.
+            fill_value = np.asarray([fill_value])[0]
```

Comment:
I'm tending to steer away from the `fill_value, = np.asarray([fill_value])` pattern as it's quite subtle.

However, we could opt for `[fill_value] = np.asarray([fill_value])` instead ...